src/fxos/reboot.js

```diff
@@ -0,0 +1,11 @@
+let adb = require('../adb');
+
+module.exports = async function reboot(hard = true) {
+  if (hard) {
+    await adb.shell('reboot');
+    await adb.waitForDevice();
+    return;
+  }
+
+  await adb.shell('stop b2g && start b2g');
```

Comment:
Sometimes, this may not be enough. If we hang during a test run and don't clean up for some reason, stop b2g will fail to shut it down as we start it via it's shell script directly from mozrunner. 

You may need to strong arm it by doing a b2g-ps first, grabbing the pids and killing all of them that way.